pyensembl/gene.py

```diff
@@ -43,40 +50,36 @@ def __str__(self):
     def __repr__(self):
         return str(self)
 
-    @property
+    @memoized_property
     def transcripts(self):
         """
         Property which dynamically construct transcript objects for all
         transcript IDs associated with this gene.
         """
-        if not hasattr(self, "_transcripts"):
-            transcript_ids_query = """
-                SELECT transcript_id
-                FROM ensembl
-                WHERE gene_id = ?
-                AND feature = 'transcript'
-            """
-            cursor = self.db.execute(transcript_ids_query, [self.id])
-            results = cursor.fetchall()
+        transcript_ids_query = """
+            SELECT transcript_id
+            FROM ensembl
+            WHERE gene_id = ?
+            AND feature = 'transcript'
+        """
+        cursor = self.db.execute(transcript_ids_query, [self.id])
+        results = cursor.fetchall()
 
-            # We're doing a SQL query for each transcript ID to fetch
-            # its particular information, might be more efficient if we
-            # just get all the columns here, but how do we keep that modular?
-            transcripts = [
-                Transcript(result[0], self.db)
-                for result in results
-            ]
-            self._transcripts = transcripts
-        return self._transcripts
+        # We're doing a SQL query for each transcript ID to fetch
+        # its particular information, might be more efficient if we
+        # just get all the columns here, but how do we keep that modular?
+        transcripts = [
+            Transcript(result[0], self.db)
+            for result in results
+        ]
+        self._transcripts = transcripts
```

Comment:
Good catch, this code actually works for arcane reasons (memoized_property will run this function to set self._transcripts and then return the value found there). Obviously need to change it to be sane.